bf2fc6cc711aee1a0c2a/kas-fleet-manager
# MGDSTRM-3644 - added common pollers

## Description
We had different timeout for the same operation in different tests.
This PR adds common pollers and use them across all the test to have a uniform polling.
A new feature that enable to dump the database has been added.
Logging has been enhanced to show more details about what is being polled.

```
1/4 [0s] - Waiting for an ID to be assigned to the cluster (&{Provider:aws Region:us-east-1 MultiAZ:false Status:})
2/4 [30s] - Waiting for an ID to be assigned to the cluster (&{Provider:aws Region:us-east-1 MultiAZ:false Status:})
3/4 [1m0s] - Waiting for an ID to be assigned to the cluster (&{Provider:aws Region:us-east-1 MultiAZ:false Status:})
3/4 [1m0s] - Polling finished
1/240 [0s] - Waiting for cluster '1l2fnpl3r8tqrlbl0rpbbu162hoii6ti' to reach status 'ready'
4/240 [1m30s] - Waiting for cluster '1l2fnpl3r8tqrlbl0rpbbu162hoii6ti' to reach status 'ready' (current status: 'cluster_provisioning')
8/240 [3m30s] - Waiting for cluster '1l2fnpl3r8tqrlbl0rpbbu162hoii6ti' to reach status 'ready' (current status: 'cluster_provisioning')
12/240 [5m30s] - Waiting for cluster '1l2fnpl3r8tqrlbl0rpbbu162hoii6ti' to reach status 'ready' (current status: 'cluster_provisioning')
...
76/240 [37m30s] - Waiting for cluster '1l2fnpl3r8tqrlbl0rpbbu162hoii6ti' to reach status 'ready' (current status: 'waiting_for_kas_fleetshard_operator')
80/240 [39m30s] - Waiting for cluster '1l2fnpl3r8tqrlbl0rpbbu162hoii6ti' to reach status 'ready' (current status: 'waiting_for_kas_fleetshard_operator')

```

Example DB Dump output:
```
+--------------------------------------+--------+--------------------------------+
|              CLUSTER ID              | STATUS |           UPDATED AT           |
+--------------------------------------+--------+--------------------------------+
| 2aad9fc1-c40e-471f-8d57-fdaecc7d3686 | ready  | 2021-06-02 14:35:33.975377     |
|                                      |        | +0200 CEST                     |
+--------------------------------------+--------+--------------------------------+
1/240 [0s] - Waiting for cluster '2aad9fc1-c40e-471f-8d57-fdaecc7d3686' to reach status 'ready'

```
:warning: Only CLUSTER table dump is currently enabled. Other dump can be easily added by calling the `DumpDB` method on the poller builder

## Verification Steps
1. Setup the development environment
2. run the tests with `TEST_SUMMARY_FORMAT=standard-verbose OCM_ENV=development make test/integration`
3. Check the logs and verify the new logs are showed
4. Tests should succed

## Type of change

- [ ] Refactor (A code change that neither fixes a bug nor adds a feature. The work is done to address technical debt) 

## Checklist (Definition of Done)
